code/modules/vr/vr_avatar.dm

```diff
@@ -16,73 +17,128 @@
 	..()
 
 /mob/living/carbon/human/virtual_reality/death()
-	if(can_die())
+	if(!can_die())
+		return FALSE
+	handle_despawn()
+	return ..(gibbed = 1)
+
+/mob/living/carbon/human/virtual_reality/proc/handle_despawn()
+	myroom.players.Remove(src)
+	vr_all_players.Remove(src)
+	if((myroom.players.len == 0) && (myroom.expires == 1))
+		if(myroom.delete_timer)
+			deltimer(myroom.delete_timer)
+		myroom.delete_timer = addtimer(CALLBACK(myroom, /datum/vr_room/proc/cleanup), 3 MINUTES, TIMER_STOPPABLE)
+	else if((myroom.players.len == 1) && (myroom.expires == 2))
+		if(myroom.round_timer)
+			deltimer(myroom.round_timer)
+		myroom.round_timer = addtimer(CALLBACK(myroom, /datum/vr_room/proc/vr_round), 10 SECONDS, TIMER_STOPPABLE)
+		var/mob/living/carbon/human/virtual_reality/winner = myroom.players[1]
+		myroom.round_end = world.time + 1 MINUTES
+		for(var/mob/living/carbon/human/virtual_reality/P in vr_all_players)
+			if(P.ckey)
+				to_chat(P, "SYSTEM MESSAGE: [winner.name] has won the PVP match in The [myroom.name]. New match in one minute, please be in the lobby.")
```

Comment:
should probably do `in [myroom]` here, instead of `The [myroom.name]` in line with contributing guidelines (and that putting `The` is just ugly IMO) 😄